### ISSUE TITLE
Link has suspicious accessible name.

### ISSUE THREAD

**Issue by anna-guo-level-access** [Anna-G-s-Org/Anna-G-s-Org-Repo#39]
**Rule description**  
One or more A elements, without an ARIA-assigned role, not intentionally hidden in the DOM and available to assistive technologies, has a suspicious calculated accessible name value.

**Tool name**  
Access Engine

**Severity**  
High

**WCAG Success criteria**  
[2.4.4 Link Purpose (In Context) - Level A](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context)

**Why it matters**  
When a link's purpose can not be determined users may be required to follow the link to determine its purpose. Returning to a previous location can often be more difficult for users with disabilities using assistive technology including those who are blind, low vision, have cognitive disabilities, or those with motor disabilities. 

**What to do**  
<p>Provide an accessible name of each link that accurately reflects the target and purpose of the link when the link and its surrounding text is taken into context. This can be done by changing text that is used to calculate a link's accessible name, or by modifying text before the link, such as in the same sentence, paragraph, list item, etc. Even when combined with text in context, it may still be helpful to users of screen readers to provide an accessible link name that makes the link's purpose clear out of context.</p>
<p>Link text that is too long, too short, or uses common terms or duplicates other link text on the same page needs to be reviewed. to determine if it accurately communicates the link purpose in context.</p>
<p>If the accessible name of the link cannot be updated, authors should ensure that the content surrounding the link aids in identifying the links purpose. Acceptable methods of providing purpose via link context includes indicating the purpose in the same:</p>
 <ol>
 <li>Sentence, paragraph, or list item or</li> 
<li>Table cell as the link or</li>
 <li>In the table header cell for a link in a data table</li>
 </ol>
 <h3>Accessible name calculation</h3> 
<p>The <a href=" https://www.w3.org/TR/html-aam-1.0/#a-element-with-a-href-attribute ">accessible name of a link</a> is calculated based on the <a href="https://www.w3.org/TR/accname-1.1/">name calculation algorithm</a>. In simplified terms, the following is used to determine the name and takes precedence rather than concatenating each method together.</p>
<ul>
  <li> <a href="http://www.w3.org/TR/2012/NOTE-WCAG20-TECHS-20120103/ARIA1"> <code>aria-labelledby</code> </a> referenced content</li>
  <li> <code>aria-label</code> </li>
  <li>element subtree including images with alternative text, and CSS off-screen positioned text</li>
  <li>the <code>title</code> attribute</li>
</ul>
It is important to remember that overly verbose link names may be difficult to understand and when located in data table header cells may create extraneous verbosity when users of screen reader navigate associated data table cells. 
<h3>Accessible Descriptions</h3> 
An accessible description can be provided by the <code>title</code> attribute or aria-describedby can be used to supplement the accessible name. However, due to how screen readers announce the accessible description, it is recommended to communicate the link purpose via the link's accessible name and/or the link's context. 
Note: The title attribute only acts as accessible name when there is no other method of calculating a name, otherwise it becomes the accessible description of the link. 
<p>When the link has an accessible name, such as one provided by text content, an image with alternative, or a label provided by ARIA, the <code>title</code> attribute acts as an accessible description. This description may not be announced by certain navigation methods, by default, by many major  screen readers, like JAWS, VoiceOver or NVDA. It may be announced when tabbing or shift+tabbing to the link but is not announced when arrowing to the link.
<p>For users who are not using a screen reader, the <code>title</code> is also displayed when the mouse is moved over a   link; however, this feature is not keyboard accessible in nearly all user agents. This is one reason why only supplementary information can be provided via this method.</p>
<h3>Links to file formats or external sites</h3>
<p>For links that indicate the type of media or an external destination, an image can also be inserted inside the link with appropriate <code>alt</code> text indicating the context (e.g., using a PDF icon to differentiate a PDF link from an HTML link). This approach works well for links that direct users to other sites or open new windows. In these cases, an icon image with appropriate <code>alt</code>text can be used to indicate the actions. WCAG 2 Level A/AA does not   require mention that links open windows or point to PDF files but does indicate it as advisory. For this best   practice, the link text in context for accessibility only need communicate what is seen visually -- if a link's  purpose is ambiguous to everyone, it is not a failure of this test.</p>
<h3>Links that are repurposed for UI elements</h3>
<p>When a link combined with a role represents a non-link UI element, such as a page tab, refer to the tests related to those controls for guidance.</p>
<h3>A Note about Off-screen positioned text</h3>
<p>Text that is positioned off-screen using CSS can be a good solution for users of screen readers; however, this text is not  accessible to anyone who has CSS turned on and is not using a screen reader. For this reason, it is recommended that this method only be used with this supplementary information is only needed by users of screen readers or where it is duplicated using other methods such as the via text in the same sentence, paragraph, list items, table cell, etc. Typically, this method works by creating a CSS style, such as <code>.sronly</code> which places the text content off-screen using CSS and then  this class is applied to a <code>span</code> element within the anchor that contains the text to hide.

**Task in the Level Access Platform**  
https://tenantb.localhost-dashboard.essentialaccessibility.com/my-tasks/6a6a1b5c6b959885e3bad8c5/6a6c09d0a63579ee96a6bf9d

**Comment by anna-guo-level-access** [comment]
Task pushed to GitHub Issues by Anna Guo from Level Access Platform.

Posted by Anna Guo via Level Access

**Comment by anna-guo-level-access** [comment]
<a href="https://tenantb.localhost-dashboard.essentialaccessibility.com/tasks/6a6c09d0a63579ee96a6bf9d/attachments/6a6c09d0a63579ee96a6bf9b/download" target="_blank" rel="noopener noreferrer">Linked Findings.csv</a>
